CMakeLists.txt

```diff
@@ -139,6 +139,14 @@ if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
 #    set(DEFAULT_MALLOC "jemalloc")
 endif()
 
+hpx_option(HPX_WITH_THREAD_STACKOVERFLOW_DETECTION
+  BOOL
+  "Enable stackoverflow detection for HPX threads/coroutines. (default: ON)"
+  ON ADVANCED)
```

Comment:
I am not sure about the overheads this detection adds. I am leaning towards only enabling it on request (If posix/linux based platform), given that prerequiste, we can also safely enable it when the build type is debug.